web-client/src/presenter/computeds/reportMenuHelper.test.ts

```diff
@@ -0,0 +1,82 @@
+import { ROLES } from '../../../../shared/src/business/entities/EntityConstants';
+import { applicationContextForClient as applicationContext } from '../../../../shared/src/business/test/createTestApplicationContext';
+import { getUserPermissions } from '../../../../shared/src/authorization/getUserPermissions';
+import { reportMenuHelper as reportMenuHeaderComputed } from './reportMenuHelper';
+import { runCompute } from 'cerebral/test';
+import { withAppContextDecorator } from '../../../src/withAppContext';
+
+describe('reportMenuHelper', () => {
+  const reportMenuHelper = withAppContextDecorator(
+    reportMenuHeaderComputed,
+    applicationContext,
+  );
+
+  const getBaseState = user => {
+    applicationContext.getCurrentUser.mockReturnValue(user);
+    return {
+      currentPage: 'CaseDetailInternal',
+      permissions: getUserPermissions(user),
+    };
+  };
+
+  describe('showActivityReport', () => {
+    it('should be true when the current user is a judge user', () => {
+      const result = runCompute(reportMenuHelper, {
+        state: getBaseState({ role: ROLES.judge }),
+      });
+
+      expect(result.showActivityReport).toBeTruthy();
+    });
+
+    it('should be true when the current user is a chambers user', () => {
+      const result = runCompute(reportMenuHelper, {
+        state: getBaseState({ role: ROLES.chambers }),
+      });
+
+      expect(result.showActivityReport).toBeTruthy();
+    });
+
+    it('should be false when the current user is NOT a judge or chambers user', () => {
+      const result = runCompute(reportMenuHelper, {
+        state: getBaseState({ role: ROLES.petitionsClerk }),
+      });
+
+      expect(result.showActivityReport).toBeFalsy();
+    });
+
+    it('should be false when permissions is undefined because the user has logged out', () => {
+      const result = runCompute(reportMenuHelper, {
+        state: {
+          ...getBaseState(undefined),
+          permissions: undefined,
+        },
+      });
+
+      expect(result.showActivityReport).toBeFalsy();
+    });
+  });
+
+  describe('pageIsReports', () => {
+    it('should show a border under the Reports tab to indicate it is the active tab when the current page is Case Deadlines Report', () => {
```

Comment:
nit: Maybe "should recognize when the current page is Case Deadlines Report" ? I don't seem to see anything here about showing a border.

Reply:
I agree with you!

I think the reason the test name was written this way was to provide a "because"/business need statement to help explain the need for that property - it's only used in the menu to determine if a class should be applied or not. That's the reason I left it, I liked that the test was explaining why the property was needed, what it's used for, in case the test ever fails we can understand why that property is needed. 

What are your thoughts on that?

Reply:
makes sense! approved.